### DISCUSSION TITLE
Graphics for Job Status

### DISCUSSION THREAD

**Discussion by Joe-Winchester** [zowe/zowe-chat#64] [Ideas]
Graphics for Job status:
Jobs have three main states.
- Waiting to be started = INPUT
- Started = ACTIVE.   (An active job can be held when it becomes HELD and RELEASED when it becomes active again).
- Finished = OUTPUT.  
  - A job that was unable to be submitted because it was not valid JCL syntax will be JCL Error.  
  - If the job crashes out it will have an ABEND status together with a reason code
  - If the job ends normally it will have a CC response, together with a return code which could be 0000,  or 0004 (for example)
  
For graphics what about a generic "Job" icon, that we can annotate to show it going through the flow.   A run arrow to show it's active,  a redX to show abended,  a warning triangle to show CC of non zero, ...  

A rough sketch of this is below (using the same icons as the IBM Explorer for z/OS).   I'm sure they can be improved on and there isn;t one for held, although this could be the Active circle with two pause vertical bars over it.  

<img width="550" alt="image" src="https://user-images.githubusercontent.com/28100302/177583198-6d51feb7-18cb-4a04-8bde-eef4acc92a12.png">

**Comment by FALLAI-Denis** [comment]
A "job" has also a type: job, TSO user, started task, or initiator.

I suggest that a different icon be associated with each type.

![image](https://user-images.githubusercontent.com/63861965/177603673-7491d075-cfc9-433c-bfa7-87220b360684.png)

However, Zowe CLI z/OSMF does not return the job type... so you would have to rely on the jobid prefix:

![image](https://user-images.githubusercontent.com/63861965/177604666-9fe0165d-f326-4dc9-84c5-030a6082e9d1.png)

![image](https://user-images.githubusercontent.com/63861965/177604979-777bad70-78a1-46b0-bf47-aec046d2c9c9.png)

**Reply by Phoebecdy91** [reply]
Hi @FALLAI-Denis Thanks for putting this up. It's really helpful! And This is something we are working on, making information more intuitive for user. It will be optimized in our latest incubator project Zowe Chat. 

If I didn't remember it wrong, you took a survey about [User Scenario Feedback for Mainframe](https://www.surveymonkey.com/r/YCCSXJG) in May. And you put up with some good points about Chatbot in Zowe. We really appreciate your time and feedback! And glad to see you in GitHub as well.

I am Phoebe, the User Experience Design Lead for IBM Zowe team. We are now collecting feedback of **Zowe Chat early stage design**. If you are interested in that, we would love to invite you to join our feedback session. It could be either way based on your interests and availability:
- **1 hour online meeting** for usability test
- **offline feedback:** we will record a video for mockup walk-through and you can leave your feedback and comments.

If there is other way that is more convenient for you, feel free to let me know. For more info of Zowe Chat: https://medium.com/zowe/zowe-gets-chatty-842e3b548902

**Comment by Joe-Winchester** [comment]
@FALLAI-Denis ,  is the way to work out the type
- JOB*   =.  Batch job
- STC*.  =    Started Task
- TSU*.   =.  TSO user
What is the prefix for an initiator ?  I've also seen SYSTASK sometimes in SDSF.  Despite all my years of z/OS tooling I'm not completely sure of the difference and also how to determine which is which kind or what they all mean.
